src/Models/UserModel.php

```diff
@@ -222,6 +222,7 @@ public function save($data): bool
                 lang('Database.emptyDataset', ['update']),
             ];
             if (in_array($e->getMessage(), $messages, true)) {
+                // @TODO Why true? Shouldn't this workaround be removed?
```

Comment:
I am not a fan of the whole `save()` extension. Lonnie had trouble getting model callbacks to work because the entity isn't passed, but I think finding another way to handle these is important (eventually).

Reply:
I think the thing to do is to fix CodeIgniter to pass the original data along in the Model callbacks, honestly. I think that would be helpful for many people.

Reply:
> to fix CodeIgniter to pass the original data along in the Model callbacks

I agree with it.

Reply:
Sounds good! I am fine with this for now.